### PR TITLE
(+semver: feature) Add push and pull functionality

### PR DESCRIPTION
**Pull request by annymsMthd**
Signed-off-by: Joshua Benjamin <joshuabnjmn@gmail.com>

### CODE REVIEW

**Comment by annymsMthd** [comment]
Code coverage is abysmal but I wanted to get the bare bones out now and get the deploy pipeline top notch.

File: cmd/idl-repository/cmd/root.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+		server := repository.NewServer(settings, storage)
+
+		ctx, cancel := context.WithCancel(context.Background())
+		grp, ctx := errgroup.WithContext(ctx)
+
+		grp.Go(server.Run())
+
+		// Wait for SIGINT/SIGTERM
+		waiter := make(chan os.Signal, 1)
+		signal.Notify(waiter, syscall.SIGINT, syscall.SIGTERM)
```
Comment (thzinc) [inline cmd/idl-repository/cmd/root.go]: This doesn't seem to be respecting Ctrl+C.

Comment (annymsMthd) [reply cmd/idl-repository/cmd/root.go]: The web server isn't setup right to cancel when the context is done. I'll fix that

File: internal/storage/fileStorage.go [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+	"github.com/pkg/errors"
+)
+
+type FileStorage struct {
+	basePath string
+}
+
+func NewFileStorage(basePath string) (*FileStorage, error) {
+	_, err := os.Stat(basePath)
```
[the remark this answers is not in the document]
Comment (annymsMthd) [reply internal/storage/fileStorage.go]: Good call

**Comment by annymsMthd** [comment]
@thzinc I'll get those changes in tonight. Thanks for the review. Good stuff

**Comment by thzinc** [comment]
@annymsMthd thanks! This is looking really cool.

**Comment by thzinc** [comment]
@annymsMthd, did you end up making any of the changes noted above? If not, I'm going to take a crack at it today because I'd like to be able to use this in my planned work.

File: cmd/idl-repository/cmd/root.go (hunk 1 above)
Comment (annymsMthd) [reply cmd/idl-repository/cmd/root.go]: This is fixed  by the changes here https://github.com/syncromatics/idl-repository/pull/1/files#diff-e16bea04351e58ac3fd0ab77080cd53dR70

**Pull request merged**